code/implementation/docker/api/src/main/java/io/cattle/platform/docker/api/ContainerLogsActionHandler.java

```diff
@@ -0,0 +1,91 @@
+package io.cattle.platform.docker.api;
+
+import io.cattle.platform.api.action.ActionHandler;
+import io.cattle.platform.archaius.util.ArchaiusUtil;
+
+
+import io.cattle.platform.core.model.Host;
+import io.cattle.platform.core.model.Instance;
+import io.cattle.platform.docker.api.model.ContainerLogs;
+import io.cattle.platform.docker.api.model.HostAccess;
+import io.cattle.platform.docker.constants.DockerInstanceConstants;
+import io.cattle.platform.docker.util.DockerUtils;
+import io.cattle.platform.host.model.HostApiAccess;
+import io.cattle.platform.host.service.HostApiService;
+import io.cattle.platform.object.ObjectManager;
+import io.cattle.platform.util.type.CollectionUtils;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import com.netflix.config.DynamicStringProperty;
+
+public class ContainerLogsActionHandler implements ActionHandler {
+
+    private static final DynamicStringProperty CONSOLE_AGENT_SCHEME = ArchaiusUtil.getString("console.agent.scheme");
+    private static final DynamicStringProperty CONSOLE_AGENT_PORT = ArchaiusUtil.getString("console.agent.port");
+    private static final DynamicStringProperty CONSOLE_AGENT_PATH = ArchaiusUtil.getString("console.agent.path");
+
+    HostApiService apiService;
+    ObjectManager objectManager;
+
+    @Override
+    public String getName() {
+        return "instance.logs";
+    }
+
+    @Override
+    public Object perform(String name, Object obj, ApiRequest request) {
+
+        Host host = null;
+        Instance container = null;
+
+        if (obj instanceof Instance) {
+            container = (Instance) obj;
+            host = DockerUtils.getHostFromContainer(objectManager, container, null);
+        }
+
+        if (host == null) {
+            return null;
+        }
+
+        ContainerLogs exec = request.proxyRequestObject(ContainerLogs.class);
+
+        Map<String, Object> data = CollectionUtils.asMap(DockerInstanceConstants.DOCKER_ATTACH_STDIN, true, DockerInstanceConstants.DOCKER_ATTACH_STDOUT, true,
```

Comment:
Most of these constants are needed. ATTACH_STDIN/STDOUT.  TTY makes no sense.  CMD???